pkg/operator2/metrics.go

```diff
@@ -0,0 +1,106 @@
+package operator2
+
+import (
+	"errors"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (c *authOperator) handleServiceMonitor(svc *v1.Service) (*monitoringv1.ServiceMonitor, err) {
+	// Do nothing if no service provided
+	if svc == nil {
+		glog.V(5).Infof("not creating ServiceMonitor for nil service")
+		return nil, nil
+	}
+	// Fetch any existing ServiceMonitor instance
+	glog.V(5).Infof("fetching ServiceMonitor %s/%s", svc.GetNamespace(), svc, GetName())
+	existing, err := c.servicemonitors(svc.GetNamespace()).Get(svc.Name())
+	if err == nil || !apierrors.IsNotFound(err) {
+		// Existing instance found, or unknown error
+		return existing, err
+	}
+
+	// No existing instance found; attempt to create default
+	glog.V(5).Infof("creating ServiceMonitor '%s/%s'", svc.GetNamespace(), svc, GetName())
+	created, err := c.servicemonitors(svc.GetNamespace()).Create(serviceMonitorForService(svc))
+	if err == nil || !apierrors.IsAlreadyExists(err) {
```

Comment:
Why refetch it? We failed creating it for whatever reason, try in next sync loop iteration?

Reply:
Also, should we maybe stomp on any changes to this monitor?